staticd/static_vty.c

```diff
@@ -1123,6 +1000,278 @@ DEFPY_YANG(ipv6_route_vrf,
+			vty_out(vty, " tag %" PRIu32, tag);
+	}
+
+	distance = yang_dnode_get_uint8(path, "./distance");
+	if (distance != ZEBRA_STATIC_DISTANCE_DEFAULT || show_defaults)
+		vty_out(vty, " %" PRIu8, distance);
+
+	iter.vty = vty;
+	iter.first = true;
+	yang_dnode_iterate(mpls_label_iter_cb, &iter, nexthop,
+			   "./mpls-label-stack/entry");
+
+	nexthop_vrf = yang_dnode_get_string(nexthop, "./vrf");
+	if (strcmp(vrf, nexthop_vrf))
+		vty_out(vty, " nexthop-vrf %s", nexthop_vrf);
+
+	table_id = yang_dnode_get_uint32(path, "./table-id");
+	if (table_id || show_defaults)
+		vty_out(vty, " table %" PRIu32, table_id);
+
+	if (yang_dnode_exists(nexthop, "./onlink")) {
+		onlink = yang_dnode_get_bool(nexthop, "./onlink");
+		if (onlink)
+			vty_out(vty, " onlink");
+	}
+
+	if (yang_dnode_exists(nexthop, "./srte-color"))
+		vty_out(vty, " color %s",
+			yang_dnode_get_string(nexthop, "./srte-color"));
+
+	vty_out(vty, "\n");
+}
+
+void static_nexthop_cli_show(struct vty *vty, struct lyd_node *dnode,
+			     bool show_defaults)
+{
+	const struct lyd_node *path = yang_dnode_get_parent(dnode, "path-list");
+	const struct lyd_node *route =
+		yang_dnode_get_parent(path, "route-list");
+
+	nexthop_cli_show(vty, route, NULL, path, dnode, show_defaults);
+}
+
+void static_src_nexthop_cli_show(struct vty *vty, struct lyd_node *dnode,
+				 bool show_defaults)
+{
+	const struct lyd_node *path = yang_dnode_get_parent(dnode, "path-list");
+	const struct lyd_node *src = yang_dnode_get_parent(path, "src-list");
+	const struct lyd_node *route = yang_dnode_get_parent(src, "route-list");
+
+	nexthop_cli_show(vty, route, src, path, dnode, show_defaults);
+}
+
+int static_nexthop_cli_cmp(struct lyd_node *dnode1, struct lyd_node *dnode2)
+{
+	enum static_nh_type nh_type1, nh_type2;
+	struct prefix prefix1, prefix2;
+	int ret = 0;
+
+	nh_type1 = yang_dnode_get_enum(dnode1, "./nh-type");
+	nh_type2 = yang_dnode_get_enum(dnode2, "./nh-type");
+
+	if (nh_type1 != nh_type2)
+		return (int)nh_type1 - (int)nh_type2;
+
+	switch (nh_type1) {
+	case STATIC_IFNAME:
+		ret = if_cmp_name_func(
+			yang_dnode_get_string(dnode1, "./interface"),
+			yang_dnode_get_string(dnode2, "./interface"));
+		break;
+	case STATIC_IPV4_GATEWAY:
+	case STATIC_IPV6_GATEWAY:
+		yang_dnode_get_prefix(&prefix1, dnode1, "./gateway");
+		yang_dnode_get_prefix(&prefix2, dnode2, "./gateway");
+		ret = prefix_cmp(&prefix1, &prefix2);
+		break;
+	case STATIC_IPV4_GATEWAY_IFNAME:
+	case STATIC_IPV6_GATEWAY_IFNAME:
+		yang_dnode_get_prefix(&prefix1, dnode1, "./gateway");
+		yang_dnode_get_prefix(&prefix2, dnode2, "./gateway");
+		ret = prefix_cmp(&prefix1, &prefix2);
+		if (!ret)
+			ret = if_cmp_name_func(
+				yang_dnode_get_string(dnode1, "./interface"),
+				yang_dnode_get_string(dnode2, "./interface"));
+		break;
+	case STATIC_BLACKHOLE:
+		/* There's only one blackhole nexthop per route */
+		ret = 0;
+		break;
+	}
+
+	if (ret)
+		return ret;
+
+	return if_cmp_name_func(yang_dnode_get_string(dnode1, "./vrf"),
```

Comment:
Should this compare not be the first compare criteria?

Reply:
I realise this is comparing the next hops within the same route. So the VRF here is the outgoing VRF the next hop points to. So maybe this correct order. I am really not sure what the order should be. Please give it a thought and take a final call appropriately.

Reply:
Yes, it's nexthop-vrf so it should be last to compare.